kiva/fonttools/font_manager.py

```diff
@@ -139,10 +139,15 @@ def set_default_weight(self, weight):
 
     def update_fonts(self, filenames):
         """ Update the font lists with new font files.
-        Currently not implemented.
         """
-        #  !!!!  Needs implementing
-        raise NotImplementedError
+        afm_paths = scan_user_fonts(filenames, fontext="afm")
+        ttf_paths = scan_user_fonts(filenames, fontext="ttf")
+
+        self.afmlist.extend(create_font_list(afm_paths))
+        self.ttflist.extend(create_font_list(ttf_paths))
+
+        self.ttf_lookup_cache = {}
+        self.afm_lookup_cache = {}
```

Comment:
Just to confirm, we're resetting these because these get used in the `findfont` method - and if we don't reset them, we end up not being able to discover the newly added font files. Right?

Reply:
That's right.

Reply:
Actually this isn't needed.